runelite-client/src/main/java/net/runelite/client/plugins/chathistory/ChatHistoryPlugin.java

```diff
@@ -269,6 +267,82 @@ else if (COPY_TO_CLIPBOARD.equals(menuOption) && !Strings.isNullOrEmpty(currentM
 		}
 	}
 
+	@Subscribe
+	public void onMenuEntryAdded(MenuEntryAdded entry)
+	{
+		String option = Text.removeTags(entry.getOption());
+		ChatboxTab tab = ChatboxTab.of(entry.getActionParam1());
+
+		if (!config.clearHistory() || tab == null || !option.equals(tab.getAfter()))
+		{
+			return;
+		}
+
+		// Public + private Clear history entries are all one word as the option.
+		// MenuEntry target is empty string.
+		final MenuEntry clearEntry = new MenuEntry();
+		ChatMessageBuilder optionBldr = new ChatMessageBuilder();
+		if (tab != ChatboxTab.ALL)
+		{
+			optionBldr.append(Color.yellow, String.format("%s: ", tab.getName()));
+		}
+		optionBldr.append(CLEAR_HISTORY);
+		clearEntry.setOption(optionBldr.build());
+
+		clearEntry.setTarget("");
```

Comment:
Empty string is the default and doesn't need to be explicitly set.

Reply:
I've tried, but `null` crashes the game (and menu swapper plugin)

```
2020-05-11 09:34:49 [Client] ERROR client-patch - Game crash: null
java.lang.NullPointerException: null
	at hz.hx(hz.java:8341)
	at client.hv(client.java:8326)
	at client.gc(client.java:8316)
	at client.ey(client.java:3142)
	at client.u(client.java:1113)
	at br.mk(br.java:336)
	at br.run(br.java:315)
	at java.base/java.lang.Thread.run(Thread.java:830)
Error: hz.hx() client.hv() client.ho() client.fj() client.au() br.ab() | hz:8341 client:8326 client:8316 client:3142 client:1113 br:336 br:315 Thread:830 | java.lang.NullPointerException
```

Reply:
Is that with `clearEntry.setTarget(null)` or with this line omitted?

Reply:
both. setting `clearEntry.setTarget(null);` and not explicitly setting the target.

Here's menu swapper's stacktrace too.

```
2020-05-11 12:23:43 [Client] WARN  n.runelite.client.eventbus.EventBus - Uncaught exception in event subscriber
java.lang.NullPointerException: null
	at java.base/java.util.regex.Matcher.getTextLength(Matcher.java:1770)
	at java.base/java.util.regex.Matcher.reset(Matcher.java:416)
	at java.base/java.util.regex.Matcher.<init>(Matcher.java:253)
	at java.base/java.util.regex.Pattern.matcher(Pattern.java:1134)
	at net.runelite.client.util.Text.removeTags(Text.java:83)
	at net.runelite.client.plugins.menuentryswapper.MenuEntrySwapperPlugin.swapMenuEntry(MenuEntrySwapperPlugin.java:428)
	at net.runelite.client.plugins.menuentryswapper.MenuEntrySwapperPlugin.onClientTick(MenuEntrySwapperPlugin.java:835)
	at net.runelite.client.eventbus.EventBus$Subscriber.invoke(EventBus.java:73)
	at net.runelite.client.eventbus.EventBus.post(EventBus.java:222)
	at net.runelite.client.callback.Hooks.post(Hooks.java:167)
	at client.pd(client.java:14714)
	at client.ey(client.java:3105)
	at client.u(client.java:1113)
	at br.mk(br.java:336)
	at br.run(br.java:315)
	at java.base/java.lang.Thread.run(Thread.java:830)
```